### DISCUSSION TITLE
What functionality do you miss in your analyses?

### DISCUSSION THREAD

**Discussion by VPetukhov** [kharchenkolab/Baysor#67] [General]
What functionality would you like to see in the package? What are the main package downsides, which you want to be improved?

**Comment by fpestana-git** [comment]
Hi VPetukhov,

Thank you for developing Baysor. It is a quite useful tool!

So I would say at the moment for me, the biggest downside is the installation of the tool. At the moment is not stable and it makes it a bit frustrating as the tool performs quite nicely.

I managed to run the package locally, on small samples, only installation using the docker version 0.5.0 (MacOS) works, so I am not able to install the latest stable version. When trying to install any version in an HPC (linux CentOS) to run bigger samples, not a single installation method worked for me. Not sure if not having root privileges (on the HPC) have an effect on that. 

Best,
Francisco

Update: managed to install Baysor 0.5.2 using Julia version 1.7.0 with compiler gcc 6.4 (the default compiler gcc 4.8.5 was not working). Then just run julia build.jl with no issues.

**Comment by VPetukhov** [comment]
Hi @fpestana-git ,
Thank you for the feedback! Do you have the same issues with the 0.6 release? There, improving the installation was my main focus.

**Comment by zsfrbkv** [comment]
Hi Viktor,

Thank you for developing Baysor -- I think it is currently one of the best tools for image-free segmentation.

I was wondering if it is possible to use Baysor for niche detection or cell typing. As far as I understood it, Baysor produces a file with a column `cluster` that signifies a cluster assigned to a given cell. However, I did not find any information on how to optimise the number of clusters in case one wants to use them for further analyses, such as cell type annotation or niche detection. Do you consider Baysor suitable for this?

Thank you and have a nice day,

Best regards,

Zaira

**Comment by rgill10** [comment]
Hello @VPetukhov!

I have been really enjoying the baysor segmentation based on cellpose inputs. I think this will truly improve cell segmentation in the spatial transcriptomics space. 

One issue I have been noticing with how baysor draws polygons is that currently when baysor uses transcripts to make cell boundaries, the transcripts used themselves do not end up getting called into the shape. This has been what I'm seeing on MERFISH data, and it is particularly easy to see on small cells with few transcripts (See examples below). It would be great if there could be some minute buffer added to all the polygon boarders, so that transcript capture is maximized on a per cell bases. 

Currently I am using baysor outputs within the vizgen VPT pipeline to partition transcripts into cells, and there is no setting to include transcripts which fall on the polygon boarder. This may be applicable to how transcripts are partitioned on other platforms as well. Would greatly appreciate any insights on how to resolve this issue or any modifications that get built into the baysor segmentation algorithm!!

![image](https://github.com/user-attachments/assets/29a2b603-e9c3-4ddb-8962-4db16f2ee7dd)
![image](https://github.com/user-attachments/assets/28b08f70-aa4a-462a-b10e-de92ba3e7ce8)
![image](https://github.com/user-attachments/assets/9140bbc1-c2f1-48d6-b474-a84914976ed8)

**Reply by VPetukhov** [reply]
Thank you, @rgill10 ! I added it in v0.7.1.
